docs/source/blocks/examples/custom-view-and-variations.md

```diff
@@ -0,0 +1,251 @@
+    "property=og:description": "Volto block with custom schema and view components using variations"
+    "property=og:title": "Volto block with custom schema and view components using variations"
+    "keywords": "Volto, React, blocks, schema, variation, view component, Plone"
+---
+
+(custom-schema-view-and-variations)=
+
+# Block with a custom schema and variations
+
+We can create a block that uses `variations`.
+A {term}`variation` is an alternative view of a block.
+This variation is shown as an additional option in the schema editor, and lets the editor change how this block is viewed.
+
+To do so, define the schema, view component, and variations, then configure the block settings.
+
+## Preparations
+
+In your Volto add-on, create a subfolder {file}`ExampleBlock05` inside the {file}`components` folder to save all the files required to create a block.
+
+## Schema
+
+Create a {file}`Schema.js` file inside the {file}`ExampleBlock05` folder, with the following contents.
+
+```js
+import messages from './messages';
+
+const Schema = ({ intl }) => {
+  return {
+    title: intl.formatMessage(messages.block05),
+    block: 'block05',
+    fieldsets: [
+      {
+        id: 'default',
+        title: intl.formatMessage(messages.default),
+        fields: ['url', 'title'],
+      },
+    ],
+
+    properties: {
+      url: {
+        title: intl.formatMessage(messages.URL),
+        widget: 'url',
+      },
+      title: {
+        title: intl.formatMessage(messages.title),
+      },
+    },
+    required: [],
+  };
+};
+
+export default Schema;
+```
+
+## Messages
+
+As you may have noted, you prepared the block for internationalization.
+{term}`Internationalization` (i18n) is the process of creating user interfaces which are suitable for different languages and cultural contexts.
+
+To do so, create a file {file}`messages.js` in the same {file}`ExampleBlock05` folder with the following contents.
+
+```js
+import { defineMessages } from 'react-intl';
+
+const messages = defineMessages({
+  block05: {
+    id: 'block05',
+    defaultMessage: 'Block 05',
+  },
+  default: {
+    id: 'default',
+    defaultMessage: 'Default',
+  },
+  URL: {
+    id: 'URL',
+    defaultMessage: 'URL',
+  },
+  title: {
+    id: 'title',
+    defaultMessage: 'Title',
+  },
+});
+
+export default messages;
+```
+
+## View component
+
+For variations, the view component needs to use the variation template to render the contents of the block.
+You can do so using the `variation` from the `props` of the block.
+
+Create a file {file}`View.jsx` in the {file}`ExampleBlock05` folder with the following contents.
+
+```jsx
+import withBlockExtensions from '@plone/volto/helpers/Extensions/withBlockExtensions';
+import cx from 'classnames';
+import React from 'react';
+
+const View = (props) => {
+  // `data` holds the values entered in the block edit form.
+  // `className` holds the CSS class names injected into this block by Volto's `styleClassNameExtenders`.
```

Comment:
@erral it would be great to split the comments in such a way that you can view it without having to use any scrollbars.
See this screen capture in order to understand what i mean i would like not to have to horizontally scroll in order to read what was the suggestion or documentation for the options used in the block config

![comments-visible](https://github.com/user-attachments/assets/1769a935-f9f8-496a-922d-8d2bf4c0cc04)